xlators/protocol/server/src/server.c

```diff
@@ -1159,6 +1159,8 @@ server_init(xlator_t *this)
                              &conf->conf_dir);
     if (ret)
         conf->conf_dir = CONFDIR;
+    else
+        ret = -1;
```

Comment:
I don't see a resource leak for `conf` in server_init(). @aspandey could you explain how this change is relevant to CID 1447088 please?

Reply:
The problem is that if GF_OPTION_INIT
fails and control reaches to "out", it might have ret=0 and in that case under "out" we will not free up the memory. So, both the changes were made to make sure that ret=-1 before calling GF_OPTION_INIT.

Reply:
But this line seems to come much after than GF_OPTION_INIT which is in line 1149. Also, the conf_dir is not present in the `this->options` dict, setting it to CONFDIR seems to be a fallback.

I think a cleaner approach would be to have a return 0 before the `out:` label. That way, we don't have to keep re-assigning -1 to ret every time a new line (that uses ret value) is added in the future. We could rename the label as `err:`.

Reply:
Yeh, that looks a better way. will change it